traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html

```diff
@@ -653,10 +653,10 @@
             </div>
 
             <div class="modal-footer">
```

Comment:
what changed here exactly? looks like you added a type=button on one button and removed it on 2 others?

Reply:
`type="button"` specifies that the `<button>` has some Javascript action that may or may not be related in any way to the `<form>` containing it. The default `type` for a `<button>` is `type="submit"` meaning that the `<button>` is responsible for submitting it's ancestor `<form>`. The upshot is that the "show/hide advanced" `<button>` would perform form validation, which it shouldn't do because you could want to see advanced features before filling out the basic stuff, and the "create"/"update" and "fulfill request" `<button>`s were not performing validation. (Because the `<form>` has no `action`, allowing `<button>`s to submit the `<form>` has no real consequence beyond input validation. And really that shouldn't matter anyway because the submit buttons are disabled until the `<form>` has no Angular errors, so on validation everything should always check out.)

Reply:
oh, i see. makes sense.